spec/features/merchants/index_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe 'index' do
+  context 'happy path: merchants exist' do
+    let!(:merchants) { RailsEngineFacade.get_merchants }
+    let!(:merchant) { merchants.sample(1)[0] }
+    it "calls the API and returns a list of merchants" do
+      visit '/merchants'
+
+      within 'div.merchants' do
+        click_link "Merchant #{merchant.id}"
+        expect(current_path).to eq("/merchants/#{merchant.id}")
+      end
+    end
+  end
+
+
+  # context 'sad path: no merchants' do
+  #   xit "returns an error message" do
+  #     allow(RailsEngineService.get_merchants).to_return({data: [], error: "No Merchants"})
```

Comment:
I think the syntax that will get you to what you are looking for is something like this:
`allow(RailsEngineService).to receive(:get_merchants).and_return({data: [], error: "No Merchants"})`

It looks like the controller and view isn't quite set up to handle this, but this is a great approach to testing to help drive the writing of the rest of the code.